laghos.cpp

```diff
@@ -1004,14 +1012,41 @@ int main(int argc, char *argv[])
                     }
                 }
 
+                if (samplerLast)
+                {
+                    samplerLast->SampleSolution(t, last_dt, S);
+                    if (samplerLast->MaxNumSamples() == windowNumSamples + (windowOverlapSamples/2))
```

Comment:
It will give an integer, and it has to be an integer (number of samples). We can either take the floor or ceiling of the number, which is a difference of one sample. It is an arbitrary choice. This choice works if the overlap is 0.